### PR TITLE
Show all user's accounts in `keybase wallet balances`

### PR DESCRIPTION
**Pull request by zapu**
It's a bit outside of our current roadmap but I convinced myself (and @mlsteele ) that this could be useful.

But I'm not sure it's the right way to do this. I went with fetching and decrypting the entire stellar bundle because I recalled that we may want to offer a possibility of private (hidden) accounts, and even if it's not implemented right now, I'd like `wallet balances` to show them as well.

This PR is a bit conflicted with my other one, so please do not look too much into this one before the other one is merged. Also I'm fine if this one is rejected if there's another thing planned for this command.

Example run:
```
» kbs wallet balances
▶ WARNING Could not load local currency exchange rate for "GBCLXO4QB4PSRKXJM5Z3YIUDR32XWWGNXH54FUKARZB3WUKJG4MDQCSK"
Balances for account GB6BRS2WFYPPI4PQNYAWYZ3CRNJHIHWNZ67T5GTSZXQYV2DC3BGGJD2G (Primary):
XLM	4.3199900 (2.89 PLN)

Balances for account GBCLXO4QB4PSRKXJM5Z3YIUDR32XWWGNXH54FUKARZB3WUKJG4MDQCSK:
XLM	2.0000000
```

cc @patrickxb

### CODE REVIEW

File: go/stellar/stellarsvc/service.go [hunk 1]
```diff
@@ -206,3 +206,21 @@ func (s *Server) WalletInitLocal(ctx context.Context) (err error) {
 	_, err = stellar.CreateWallet(ctx, s.G())
 	return err
 }
+
+func (s *Server) WalletGetPublicKeys(ctx context.Context) (ret []stellar1.BundleEntry, err error) {
```
Comment (mlsteele) [inline go/stellar/stellarsvc/service.go]: This use of `BundleEntry` I don't like. How about `GetOwnAccountIDs(ctx,g) -> ([]stellar1.AccountID, error)`?

File: protocol/avdl/stellar1/local.avdl [hunk 2]
```diff
@@ -31,4 +31,6 @@ protocol local {
   boolean ownAccountLocal(AccountID accountID);
 
   void importSecretKeyLocal(SecretKey secretKey, boolean makePrimary);
+
+  array<BundleEntry> walletGetPublicKeys();
```
Comment (mlsteele) [inline protocol/avdl/stellar1/local.avdl]: How about `array<AccountID> OwnAccountsLocal()`?

File: go/client/cmd_wallet_balances.go [hunk 3]
```diff
@@ -47,20 +45,52 @@ func (c *cmdWalletBalances) Run() error {
 		return err
 	}
 
-	accountID := stellar1.AccountID(c.address)
-
-	balances, err := cli.BalancesLocal(context.Background(), accountID)
-	if err != nil {
-		return err
+	var accounts []stellar1.BundleEntry
```
Comment (mlsteele) [inline go/client/cmd_wallet_balances.go]: []stellar1.AccountID

File: go/client/cmd_wallet_balances.go [hunk 4]
```diff
@@ -30,48 +30,90 @@ func newCmdWalletBalances(cl *libcmdline.CommandLine, g *libkb.GlobalContext) cl
...
+
+	dui := c.G().UI.GetDumbOutputUI()
+	balances, err := cli.BalancesLocal(context.Background(), accountID)
 	if err != nil {
 		return err
 	}
 
-	accountID := stellar1.AccountID(c.address)
+	for _, balance := range balances {
+		kind := balance.Asset.Type
```
Comment (mlsteele) [inline go/client/cmd_wallet_balances.go]: This is like "credit_alphanum4". You probably want Asset.Code and Asset.Issuer

File: go/client/cmd_wallet_balances.go [hunk 5]
```diff
@@ -30,48 +30,90 @@ func newCmdWalletBalances(cl *libcmdline.CommandLine, g *libkb.GlobalContext) cl
...
 		}
-		localValue := ""
-		if localBalance.Currency != "" && localBalance.Value != "" {
-			localValue = fmt.Sprintf(" (%s %s)", localBalance.Value, localBalance.Currency)
+		dui.Printf("Balances for account %s:\n", accountName)
+
+		for _, balance := range acc.Balance {
+			localAmountStr := ""
+			kind := balance.Asset.Type
+			if balance.Asset.Type == "native" {
```
Comment (mlsteele) [inline go/client/cmd_wallet_balances.go]: Asset.Type and IsNativeXLM again

File: go/stellar/stellarsvc/service.go [hunk 6]
```diff
@@ -254,3 +217,68 @@ func (s *Server) SetDisplayCurrency(ctx context.Context, arg stellar1.SetDisplay
 	}
 	return remote.SetAccountDefaultCurrency(ctx, s.G(), arg.AccountID, arg.Currency)
 }
+
+type exchangeRateMap map[string]remote.XLMExchangeRate
+
+func getLocalCurrencyAndExchangeRate(ctx context.Context, g *libkb.GlobalContext, account *stellar1.LocalOwnAccount, rates exchangeRateMap) error {
```
Comment (mlsteele) [inline go/stellar/stellarsvc/service.go]: Comment that this mutates account

File: go/client/cmd_wallet_balances.go [hunk 7]
```diff
@@ -30,48 +30,90 @@ func newCmdWalletBalances(cl *libcmdline.CommandLine, g *libkb.GlobalContext) cl
...
+	dui := c.G().UI.GetDumbOutputUI()
+	balances, err := cli.BalancesLocal(context.Background(), accountID)
 	if err != nil {
 		return err
 	}
 
-	accountID := stellar1.AccountID(c.address)
+	for _, balance := range balances {
+		kind := balance.Asset.Type
+		if balance.Asset.Type == "native" {
```
[the remark this answers is not in the document]
Comment (zapu) [reply go/client/cmd_wallet_balances.go]: thanks a lot! didn't know about this. Should I change another occurrence of this in https://github.com/keybase/client/blob/master/go/stellar/stellar.go#L116 ?

File: go/client/cmd_wallet_balances.go [hunk 8]
```diff
@@ -30,48 +30,90 @@ func newCmdWalletBalances(cl *libcmdline.CommandLine, g *libkb.GlobalContext) cl
...
-		if asset.Type == "native" {
-			kind = "XLM"
+	for i, acc := range accounts {
+		var accountName string
+		if acc.Name != "" {
+			accountName = fmt.Sprintf("%s (%s)", acc.Name, acc.AccountID.String())
+		} else {
+			accountName = acc.AccountID.String()
+			if acc.IsPrimary {
+				accountName += " (Primary)"
```
[the remark this answers is not in the document]
Comment (zapu) [reply go/client/cmd_wallet_balances.go]: Yes, however I was typing out that decision tree here when I realized you can't set name yet (or can you)? So I left it as it is for now. I'll improve it though!

[the remark this answers is not in the document]
Comment (mlsteele) [reply go/client/cmd_wallet_balances.go]: That's true you can't yet

File: go/client/cmd_wallet_balances.go (hunk 7 above)
[the remark this answers is not in the document]
Comment (mlsteele) [reply go/client/cmd_wallet_balances.go]: Yeah that would be great

**Comment by zapu** [comment]
Changed some things around, example output for balances for an account which has several assets:
```
» kbs wallet balances GCBXDNK6ER3IWQ3IREGQHV4MBDB3GGQM4ZHTXB6CDRKTPGE2PLKNIC5T
"KEYZ"	1.0000000	(issued by GDVDSIDVX63LKSUFTO5NQ3YHPDDAYUSEPOJLFXVWCVCLEKMOERGHZ7DR)
"HUG"	1.0000000	(issued by GD5T6IPRNCKFOHQWT264YPKOZAWUMMZOLZBJ6BNQMUGPWGRLBK3U7ZNP)
XLM	426.8524200
```

Patrick mentioned that we don't need balances command that will display arbitrary account's balances, we just need one to display user's account balance. But we might as well leave this be for now and come back to this command later.

**Comment by zapu** [comment]
cc @patrickxb do you want to take a look?

**Pull request merged**